### PR TITLE
Add support for protected properties.

### PR DESCRIPTION
**Pull request by chrene**
Protected properties allows you to define properties for an object which should not be included when nested. For instance if you have an Author object and a Book object. A book has a relation to with Author, and book is public API. Author will have personal information such as social security number etc, and they can now be "protected" such that anyone looking up the author of the book will not get those information back.

tldr: It's like the "hidden" feature, but it's only hidden when nested

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by raymondfeng** [comment]
Would the `protected` flag applies to included models too? `include` can be used to pull in related models over relations. See http://docs.strongloop.com/display/LB/Include+filter.

**Comment by chrene** [comment]
I'm not sure yet, I'll test it out
